props/blaster_BC_buttons.h

```diff
@@ -0,0 +1,787 @@
+        return true;
+      case EVENTID(BUTTON_MODE_SELECT, EVENT_FIRST_SAVED_CLICK_SHORT, 0):  // mode s1
+        exit();
+        return true;
+    }
+    return SPEC::SelectCancelMode::mode_Event2(button, event, modifiers);
+  }
+};
+
+template<class SPEC>
+struct BCBlasterMenuSpec {
+  typedef BCBlasterVolumeMode<SPEC> BCBlasterVolumeMenu;
+  typedef mode::SelectCancelMode SelectCancelMode;
+  typedef mode::SteppedMode<SPEC> SteppedMode;
+  typedef mode::SteppedModeBase<SPEC> SteppedModeBase;
+  typedef mode::MenuBase<SPEC> MenuBase;
+  typedef SoundLibraryV2 SoundLibrary;
+};
+
+
+class BlasterBCButtons : public Blaster {
+public:
+  BlasterBCButtons() : Blaster() {}
+  const char* name() override { return "BlasterBCButtons"; }
+
+  void Setup() override {
+    MKSPEC<BCBlasterMenuSpec>::SoundLibrary::init();
+  }
+
+  void EnterVolumeMenu() {
+    pushMode<MKSPEC<BCBlasterMenuSpec>::BCBlasterVolumeMenu>();
+  }
+
+  void Fire() override {
+    if (DoJam()) return;
+    if (blaster_mode != MODE_AUTO && DoEmpty()) return;
+
+    switch (blaster_mode) {
+      case MODE_STUN:
+        SaberBase::DoEffect(EFFECT_STUN, 0);
+        shots_fired_++;
+        PVLOG_NORMAL << "******** STUN - Remaining shots = " << GetBulletCount() << "\n";
+        break;
+      case MODE_KILL:
+        SFX_blast.Select(-1);
+        SaberBase::DoEffect(EFFECT_FIRE, 0);
+        shots_fired_++;
+        PVLOG_NORMAL << "******** FIRE - Remaining shots = " << GetBulletCount() << "\n";
+        break;
+      case MODE_AUTO:
+        if (!CheckEmpty()) {
+          SelectAutoFirePair();
+          auto_firing_ = true;
+          SaberBase::SetLockup(LOCKUP_AUTOFIRE);
+          SaberBase::DoBeginLockup();
+        }
+        break;
+    }
+  }
+
+ void DoAutoFire() override {
+    if (!SFX_auto) return;
+    // First time through, let bgnauto finish first if we have one
+    if (SFX_bgnauto && GetWavPlayerPlaying(&SFX_bgnauto)) return;
+
+    // No bgnauto or bgnauto has finished playing, so initialize and start auto-firing
+    if (auto_firing_ && !auto_player_) {
+      auto_player_ = GetWavPlayerPlaying(&SFX_auto);
+      shots_fired_++;
+      auto_time_ = millis();
+      PVLOG_NORMAL << "******** AUTOFIRING - Remaining shots = " << GetBulletCount() << "\n";
+    }
+    // use auto.wav's length to count shots
+    if (auto_firing_ && auto_player_ && (millis() - auto_time_) > auto_sound_length) {
+      auto_sound_length = auto_player_->length() * 1000;
+      shots_fired_++;
+      if (CheckEmpty()) {
+        SaberBase::DoEndLockup();
+        SaberBase::SetLockup(SaberBase::LOCKUP_NONE);
+        auto_firing_ = false;
+        auto_player_.Free();
+        auto_time_ = millis(); // Reset for the first empty sound
+        return;
+      }
+      auto_time_ = millis();
+      PVLOG_NORMAL << "******** AUTOFIRING - Remaining shots = " << GetBulletCount() << "\n";
+    }
+
+    // This will mimic the hammer running on no bullets at the same rate it was firing.
```

Comment:
This is cool, but only guns with electric ignition would do this right?
Most auto-firing guns would only do this once.
Maybe a define?

Reply:
Well, If it's up to me because it's the BC prop, I'd like to leave it like this. I like it and have pride in the auto-timing thing, plus I feel like more sounds playing the better instead of one "click - you're empty".
Also, it's a long time ago in a galaxy far, far away. Who knows how blasters actually worked? :)